### PR TITLE
Feat/pro 1539/list devices in group

### PR DESCRIPTION
**Pull request by CalumMcCall**

### CODE REVIEW

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala [hunk 1]
```diff
@@ -35,15 +36,22 @@ class GroupsResource(namespaceExtractor: Directive1[Namespace])
             _       <- GroupMember.createGroup(groupId, namespace, request.device1)
             _       <- GroupMember.createGroup(groupId, namespace, request.device2)
                  //PRO-1378 Add logic to find devices which should be in this group
-          } yield ()
+          } yield groupId
```
Comment (Danten) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: This change seems unrelated?

Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: No, this needs to be returned as group queries require the use of the groupId, and if you just give a name when you make the request, then you won't know what the id is. It's used in the tests for example.

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupMember.scala [hunk 2]
```diff
@@ -37,4 +38,10 @@ object GroupMember {
       : DBIO[Int] =
     groupMembers += GroupMember(groupId, namespace, deviceId)
 
+  def listDevicesInGroup(groupId: Uuid, namespace: Namespace)(implicit ec: ExecutionContext): DBIO[Seq[Uuid]] =
```
Comment (Danten) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupMember.scala]: groupId is supposed to be unique so we shouldn't need to have a Namespace.

File: device-registry/src/test/scala/org/genivi/sota/device_registry/ResourceSpec.scala [hunk 3]
```diff
@@ -4,12 +4,14 @@
...
-import akka.http.scaladsl.server.{Directives, Route}
+import akka.http.scaladsl.model.Uri.Path.{Segment, Slash}
+import akka.http.scaladsl.server.{Directive1, Directives, Route}
 import akka.http.scaladsl.testkit.RouteTestTimeout
 import akka.http.scaladsl.testkit.ScalatestRouteTest
 import cats.data.Xor
+import org.genivi.sota.rest.Validation._
 import org.genivi.sota.core.DatabaseSpec
-import org.genivi.sota.data.Namespace
+import org.genivi.sota.data.{Namespace, Uuid}
```
Comment (Danten) [inline device-registry/src/test/scala/org/genivi/sota/device_registry/ResourceSpec.scala]: These imports looks weird, there is no change in the code?

File: common/src/main/scala/org/genivi/sota/http/UuidDirectives.scala [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.genivi.sota.http
+import akka.http.scaladsl.server.{Directive1, Directives}
+import org.genivi.sota.data.Uuid
+import org.genivi.sota.rest.Validation._
+
+object UuidDirectives extends Directives {
```
Comment (Danten) [inline common/src/main/scala/org/genivi/sota/http/UuidDirectives.scala]: Why do you extend Directives?

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupMember.scala (hunk 2 above)
Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupMember.scala]: Good catch.

**Comment by CalumMcCall** [comment]
Ok, I think I've addressed everything.

**Comment by Danten** [comment]
Review Score: 11/11

You need to restart the ci tests since Alex seems to have messed with teamcity.

**Comment by Danten** [comment]
Make sure to squash the commits!
